src/qt/walletmodel.cpp

```diff
@@ -79,6 +79,10 @@ void WalletModel::updateStatus()
 
 void WalletModel::pollBalanceChanged()
 {
+    // Do not even try to poll if shutdown was requested.
+    if (m_node.shutdownRequested())
```

Comment:
In commit "[WalletModel] Stop polling balance if shutdown was requested." (ae5775a16e85144e73e999ef87dec26e22236a86)

Is it necessary to include this change in the PR? I think it'd be nicer to drop this commit if it's not strictly necessary. Reasons:

- I think it is cleaner to cancel the poll timer outright when shutdown is initiated instead of continuing to poll but just returning immediately each poll interval. I already have a change which does this out for review: 2bc9b92ed8b7736ad67876398a0bb8287f57e9b3 from #18587. I think that approach is better because it lets the GUI just handle shutdown one place in qt/bitcoin.cpp, and not be querying for shutdown state in other parts of GUI code.
- Adding the m_node method call here is bad for performance and debugging with #10102 because m_node method calls require IPC between gui and node processes

If we can drop this commit from the PR, that'd be great. But it's ok to keep this if it's needed to avoid crashes. I can always revert it as part of #18587 later. (I'd appreciate review on #18587, by the way, if anyone wants to take a look.)

Reply:
Yeah absolutely, will drop it. it's not necessary anymore and fully agree on the timer cancel from outside and performance reasons (and well, because it's an ugly patch too..).

If my memory doesn't fail me, i made it because there was a race condition over the shutdown process before (for the first commit of this PR, the one that was removed due #17905 merge) and even closing the timer properly as you did there was not good enough.